operator/pkg/ingress/ingress.go

```diff
@@ -223,9 +233,31 @@ func getIngressClassName(ingress *slim_networkingv1.Ingress) *string {
 	return ingress.Spec.IngressClassName
 }
 
+func hasEmptyIngressClass(ingress *slim_networkingv1.Ingress) bool {
+	className := getIngressClassName(ingress)
+
+	return className == nil || *className == ""
+}
+
+func (ic *Controller) isCiliumIngressEntry(ingress *slim_networkingv1.Ingress) bool {
+	className := getIngressClassName(ingress)
+
+	if (className == nil || *className == "") && ic.isDefaultIngressClass {
+		return true
+	}
+
+	return className != nil && *className == ciliumIngressClassName
+}
+
 func (ic *Controller) handleIngressAddedEvent(event ingressAddedEvent) error {
-	ingressClass := getIngressClassName(event.ingress)
-	if ingressClass == nil || *ingressClass != ciliumIngressClassName {
+	if !ic.isCiliumIngressEntry(event.ingress) {
+		// this could have been our class before we should clean up
+		err := ic.garbageCollectOwnedResources(event.ingress)
+		if err != nil {
+			return err
+		}
+
+		log.WithField(logfields.Ingress, event.ingress.Name).Debug("Skipping ingress as it is not the cilium class or default")
```

Comment:
very minor nit: here (and everywhere else), always log namespace + name. Otherwise debugging things can get very tricky. People love to have the same name in multiple namespaces.

Reply:
Ah good to know, will add it